pkg/proxy/proxy.go

```diff
@@ -149,6 +168,78 @@ func (p *Proxy) StartProxy() *http.Server {
 	return srv
 }
 
+// unsecured returns true if the request does not require authentication
+func unsecured(ctx echo.Context) bool {
+	uri := ctx.Request().URL.RequestURI()
+	return uri == proxyHealthEndpoint || uri == "/.well-known/oauth-authorization-server" || strings.HasPrefix(uri, authEndpoint)
+}
+
+// auth handles requests to SSO. Used by web login.
+func (p *Proxy) auth(ctx echo.Context) error {
+	req := ctx.Request()
+	targetURL, err := url.Parse(configuration.GetRegistrationServiceConfig().Auth().SSOBaseURL())
+	if err != nil {
+		return err
+	}
+	targetURL.Path = req.URL.Path
+	targetURL.RawQuery = req.URL.RawQuery
+
+	return p.handleSSORequest(targetURL)(ctx)
+}
+
+// oauthConfiguration handles requests to oauth configuration and proxies them to the corresponding SSO endpoint. Used by web login.
+func (p *Proxy) oauthConfiguration(ctx echo.Context) error {
+	targetURL, err := url.Parse(ssoWellKnownTarget())
+	if err != nil {
+		return err
+	}
+	return p.handleSSORequest(targetURL)(ctx)
+}
+
+// openidAuth handles requests to the openID Connect authentication endpoint. Used by web login.
+func (p *Proxy) openidAuth(ctx echo.Context) error {
+	targetURL, err := url.Parse(authorizationEndpointTarget())
+	if err != nil {
+		return err
+	}
+	targetURL.Path = ctx.Request().URL.Path
+	targetURL.RawQuery = ctx.Request().URL.RawQuery
+
+	return p.redirectTo(ctx, targetURL.String())
+}
+
+func (p *Proxy) redirectTo(ctx echo.Context, to string) error {
+	log.InfoEchof(ctx, "redirecting %s to %s", ctx.Request().URL.String(), to)
+	http.Redirect(ctx.Response().Writer, ctx.Request(), to, http.StatusSeeOther)
```

Comment:
Just curious, is there a reason for using `http.StatusSeeOther` specifically here as oppose to `http.StatusFound` for example?

Reply:
302 would probably work just fine but I think 303 is more suitable here. But I don't have any strong opinion though. Those redirect options is often a bit confusing due to some overlapping :)